synapseutils/sync.py

```diff
@@ -975,3 +975,49 @@ def _check_size_each_file(df):
             single_file_size = os.stat(os.path.expandvars(os.path.expanduser(file_path))).st_size
             if single_file_size == 0:
                 raise ValueError("File {} is empty, empty files cannot be uploaded to Synapse".format(file_name))
+
+
+def generate_sync_manifest(syn, directory_path, parent_id, manifest_path):
+    """Generate manifest for syncToSynapse() from a local directory."""
+    manifest_cols = ["path", "parent"]
+    manifest_rows = _walk_directory_tree(syn, directory_path, parent_id)
+    _write_manifest_data(manifest_path, manifest_cols, manifest_rows)
+
+
+def _create_folder(syn, name, parent_id):
+    """Create Synapse folder."""
+    entity = {
```

Comment:
Look for a child with the same name first?

Reply:
Not totally necessary, it will just "update" the folder entity